scanpy/plotting/_stacked_violin.py

```diff
@@ -461,38 +475,66 @@ def _make_rows_of_violinplots(
         # All columns should have a unique name, yet, frequently
         # gene names are repeated in self.var_names,  otherwise the
         # violin plot will not distinguish those genes
-        _matrix.columns = [f"{x}_{idx}" for idx, x in enumerate(_matrix.columns)]
-
-        # transform the  dataframe into a dataframe having three columns:
+        # _matrix.columns = [f"{x}_{idx}" for idx, x in enumerate(_matrix.columns)] # added warning as this row was commented out
+        if _matrix.columns.value_counts().max() > 1:
+            warn(
+                "Duplicate gene or condition labels present."
+                "Results might be unexpected."
+            )
```

Comment:
@yugeji commenting this out doesn’t work with newer pandas versions anymore. Does your branch handle duplicated columns? Then we can just move this into the `if len(self.groupby_cols) < 1:` branch